packages/react/src/components/table/table-row.tsx

```diff
@@ -129,35 +165,19 @@ export const TableRow = <T extends object>({
     onClick,
     row,
     striped,
-    viewIndex,
 }: TableRowProps<T>): ReactElement => { // eslint-disable-line arrow-body-style
     return (
         <StyledTableRow
             clickable={!!onClick}
             data-testid={`table-row-${row.index}`}
             error={error}
-            selected={row.isSelected}
+            key={row.id}
             striped={striped}
+            selected={row.getIsSelected()}
             onClick={() => onClick && onClick(row)}
-            {...row.getRowProps() /* eslint-disable-line react/jsx-props-no-spreading */}
-            {...(onClick ? { tabIndex: 0, role: 'button' } : {}) /* eslint-disable-line react/jsx-props-no-spreading */}
+            {...(onClick ? { tabIndex: 0, role: 'button' } : {})/* eslint-disable-line react/jsx-props-no-spreading */}
```

Comment:
Avoid put comment in between code. 
It's fine to spread null and the intent is clearer.
```
onClick={() => onClick && onClick(row)}
// eslint-disable-next-line react/jsx-props-no-spreading
{...(onClick ? { tabIndex: 0, role: 'button' } : null)}
```